### ISSUE TITLE
cuda test error_atomics_gpu_mutex fails

### ISSUE THREAD

**Issue by Infinoid** [halide/Halide#5012]
The `error_atomics_gpu_mutex` test started failing for me after #4997 was merged, a couple days ago.

```
build$ HL_JIT_TARGET=host-cuda ctest -V -R error_atomics_gpu_mutex
UpdateCTestConfiguration  from :/home/infinoid/workspace/halide/build/DartConfiguration.tcl
UpdateCTestConfiguration  from :/home/infinoid/workspace/halide/build/DartConfiguration.tcl
Test project /home/infinoid/workspace/halide/build
Constructing a list of tests
Done constructing a list of tests
Updating test list for fixtures
Added 0 tests to meet fixture requirements
Checking test dependency graph...
Checking test dependency graph end
test 348
    Start 348: error_atomics_gpu_mutex

348: Test command: /home/infinoid/workspace/halide/build/test/error/error_atomics_gpu_mutex
348: Test timeout computed to be: 10000000
348: Success!
1/1 Test #348: error_atomics_gpu_mutex ..........***Failed  Required regular expression found. Regex=[Success!]  1.96 sec

0% tests passed, 1 tests failed out of 1

Label Time Summary:
error    =   1.96 sec*proc (1 test)

Total Test time (real) =   1.99 sec

The following tests FAILED:
	348 - error_atomics_gpu_mutex (Failed)
Errors while running CTest
```

Git commit 941a576f8326e5846717f6ea11b70af8128d0fdf fails.
Git commit b126ef7c2a611fffa0bf4d79249c62a18f90a356 succeeds.

I am using cuda 10.2, gcc 7.5.0 on linux amd64.

**Comment by alexreinking** [comment]
I can repro this on CUDA 10.2, MSVC 16.6.1, on Windows x64

**Comment by abadams** [comment]
Will fix. Something that used to error out can now compile correctly, so there's no actual breakage. I'll just adjust the test to be harder.

**Comment by Infinoid** [comment]
With @abadams fix (commit 01b1a14f5213213f083d03d283f9eac43d88e2a3), the test passes again for me.
